### PR TITLE
 feat(payments): Update PayPal checkout success screen to match latest UX spec

### PR DESCRIPTION
**Pull request by biancadanforth**
## Because

- The latest [success screen design specification](https://www.figma.com/file/ejDMYaKpm5bsYL5YvhItur/Payment-Integration?node-id=519%3A5800) differed from what was on `main` in a few notable ways.
- The Checkout screens on mobile don't match the current mobile spec

## This pull request
- Update the en-US `SubscriptionSuccess`screen to match the spec which includes:
  - Add the "Subscription confirmation" header and "30-day money-back guarantee" subheader to `PaymentConfirmation`.
    - Create a new `SubscriptionTitle` component, since it is also used in `PaymentProcessing` and `SubscriptionCreate`.
  - Replace "Thank You!" with "Thank you!"
  - Update message underneath "Thank you!".
  - Get the `SubscriptionTitle` and `PlanDetails` components to display per the spec in mobile.
  - Add a "success with PayPal" story to the Product route and rename the previous "success" route to "success with Stripe".
- Add the `SubscriptionTitle` component to the `PaymentProcessing` component.
- BONUS: Update the mobile view of `SubscriptionSuccess`, `SubscriptionCreate` and `PaymentProcessing` to match the mobile spec (more closely, anyway).

## Issue that this pull request solves

Closes: #7569 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### SubscriptionSuccess
Before (desktop)
BEFORE_SubscriptionSuccess_desktop

After (desktop)
AFTER_SubscriptionSuccess_desktop_2

Before  (mobile)
BEFORE_SubscriptionSuccess_mobile

After (mobile)
AFTER_SubscriptionSuccess_mobile_2

### SubscriptionCreate
Before (desktop)
BEFORE_SubscriptionCreate_desktop

After (desktop)
AFTER_SubscriptionCreate_desktop_2

Before  (mobile)
BEFORE_SubscriptionCreate_mobile

After  (mobile)
AFTER_SubscriptionCreate_mobile_2

### SubscriptionCreate > PaymentProcessing
Before (desktop)
BEFORE_PaymentProcessing_desktop

After (desktop)
AFTER_PaymentProcessing_desktop_2

Before (mobile)
BEFORE_PaymentProcessing_mobile

After (mobile)
AFTER_PaymentProcessing_mobile_2

### CODE REVIEW

**Comment by biancadanforth** [comment]
Tests are passing, but `Jest: "global" coverage threshold for branches (94%) not met: 93.84%`...

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 1]
```diff
@@ -55,82 +52,89 @@ export const PaymentConfirmation = ({
...
+        data-testid="payment-confirmation"
+      >
+        <header>
+          <img
+            className="circled-check"
+            src={circledCheckbox}
+            alt="circled checkbox"
+          />
+          <Localized id="payment-confirmation-thanks-heading">
+            <h2></h2>
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:68]: I'd prefer to have some fallback content here.

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 2]
```diff
@@ -55,82 +52,89 @@ export const PaymentConfirmation = ({
...
+          </Localized>
           <Localized
-            id="payment-confirmation-invoice-number"
-            vars={{ invoiceNumber }}
+            id="payment-confirmation-thanks-subheading"
+            vars={{ email, product_name }}
           >
             <p></p>
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:74]: Same with fallback content here.

File: packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Localized } from '@fluent/react';
+
+import './index.scss';
+
+type SubscriptionTitleProps = {
+  screenType: string;
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx:7]: Would be nice to have a sum type for this.  This type is telling me that I can pass in any string, which I don't think is true.

File: packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export const SubscriptionTitle = ({ screenType }: SubscriptionTitleProps) => {
+  const defaultHeaders: { [index: string]: string } = {
+    create: 'Set up your subscription',
+    success: 'Subscription confirmation',
+    processing: 'Confirming subscription...',
+  };
+  return (
+    <div className="subscription-title" data-testid="subscription-title">
+      <Localized id={`subscription-${screenType}-title`}>
+        <h3 className="title">{defaultHeaders[screenType]}</h3>
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx:19]: The sum type I mentioned above would help here.  `screenType` is used without anything checking that it is one of the keys of `defaultHeaders`.

**Review by chenba — COMMENTED** [review]
Would like fallback content for a number of spots.

The type for `screenType` can be narrowed down to the accepted set of strings.

I can take another look once #7613 is merged.

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 5]
```diff
@@ -55,82 +52,89 @@ export const PaymentConfirmation = ({
...
-            vars={{
-              amount: getLocalizedCurrency(amount, currency),
-              intervalCount: interval_count,
-            }}
-          >
-            <p>{planPrice}</p>
+        {Provider.isStripe(payment_provider) && (
+          <div className="billing-info" data-testid="billing-info">
+            <Localized id="payment-confirmation-billing-heading">
+              <h3></h3>
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:96]: Fallback content?

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx [hunk 6]
```diff
@@ -55,82 +52,89 @@ export const PaymentConfirmation = ({
...
+            <div className="bottom-row">
+              {displayName ? <p>{displayName}</p> : null}
+              <p>{email}</p>
+            </div>
+          </div>
+        )}
+
+        <div className="payment-details">
+          <Localized id="payment-confirmation-details-heading">
+            <h3></h3>
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:107]: Fallback content?

File: packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+type SubscriptionTitleProps = {
+  screenType: 'create' | 'success' | 'processing';
+  className?: string;
+};
+
+export const SubscriptionTitle = ({
+  screenType,
+  className = '',
+}: SubscriptionTitleProps) => {
+  const defaultHeaders: { [index: string]: string } = {
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx:15]: The type here can be `{ K in SubscriptionTitleProps['screenType']: string }` if the three keys are meant to match the three values for `screenType`.

**Review by chenba — APPROVED** [review]
LGTM

File: packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx (hunk 5 above)
Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/PaymentConfirmation/index.tsx:96]: Interesting -- this actually isn't something I touched in my patch; my local development environment is reformatting this file and committing the reformat... I can update it though...

File: packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx (hunk 7 above)
Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx:15]: This throws a `prettier` error on the git commit hook...

Comment (chenba) [reply packages/fxa-payments-server/src/components/SubscriptionTitle/index.tsx:15]: Oh sorry, yeah missed some brackets there.

**Comment by jess-cook03** [comment]
@biancadanforth as per my message Friday I wanted to make sure you got the updated mobile specs that extends the background to reflect the desktop specs. Let me know if you have questions.

Here is the link to the updated mobile spec ---> https://www.figma.com/file/ejDMYaKpm5bsYL5YvhItur/Payment-Integration?node-id=757%3A200

Thanks!

**Comment by biancadanforth** [comment]
Thanks Jess; yes I did make the title white as in the desktop specs on mobile.

**Pull request merged**